docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+      B23 --> B24["Send WCP1Hello to iframe"]
+      B24 --> B3["Receive WCP3Handshake with MessagePort"]
+      B3 --> B31["stop timeout"]
+      B11 --> B3
+      B31 --> B32["Send WCP4ValidateIdentity on MessagePort"]
+      B32 --> B321["Receive WCP5ValidateIdentityResponse"]
+      B321 --> B3211["Create DesktopAgentProxy to process DACP messages"]
+      B3211 --> B3212["Save DesktopAgentDetails to SessionStorage"]
+      B32 --> B322["Receive WCP5ValidateIdentityFailedResponse"]
+      
+      A3 --> T1["Set timeout"]
+      T1 --> T2["timeout expires"]
+      T2 --> T3{"Was a failover fn provided"}
+      T3 -->|yes|T31["Run failover"]
+      T31 --> T311{"Check failover return type"}
+      T311 -->|WindowProxy|T3111["Send WCP1Hello via WindowProxy"]
+      T311 -->|DesktopAgent|T3112["Save DesktopAgentDetails to SessionStorage"]
+      T3111 --> B3
+    end
+    P21 -->P22(["resolve with window.fdc3"])
+    P33 -->P34(["resolve with window.fdc3"])
+    B3212 --> B3213(["resolve with DesktopAgentProxy"])
+    B322 --> B3221(["reject with AgentError.AccessDenied"])
+    T3112 --> T31121(["resolve with DesktopAgent"])
+    T3 -->|no|T32(["reject with AgentError.AgentNotFound"])
+    T311 -->|other|T3113["reject with AgentError.InvalidFailover"]
+```
+
+## Providing Channel Selector and Intent Resolver UIs
+
+Users of FDC3 Desktop Agents often need access to UI controls that allow them to select user channels or to resolve intents that have multiple resolution options. Whilst apps can implement these UIs on their own via data and API calls provided by the `DesktopAgent` API, Desktop Agents typically provide these interfaces themselves.
+
+However, Browser Resident Desktop Agents may have difficulty displaying user interfaces over applications for a variety of reasons (inability to inject code, lack of permissions to display popups etc.), or may not (e.g. because they render applications in iframes within windows they control and can therefore display content over the iframe). The Web Connection Protocol and the `getAgent()` implementation based on it and incorporated into apps via the [`@finos/fdc3` npm module](https://www.npmjs.com/package/@finos/fdc3), is intended to help Desktop Agents deliver these UIs where necessary.
+
+```mermaid
+flowchart LR
+  subgraph DA [Desktop Agent Window]
+    A[(Desktop Agent)]
+  end
+  A-->B["getAgent()"]
+  subgraph App [App Window]
+    B
+    subgraph iframe1 [iframe 1]
+      cs[Channel Selector]
+    end
+    subgraph iframe2 [iframe 2]
+      ir[Intent Resolver]
+    end
+  end
+  B-->cs
+  B-->ir
+```
+
+The WCP allows applications to indicate to the `getAgent()` implementation whether they need the UIs (they may not need one or the other based on their usage of the FDC3 API, or because they implement UIs themselves) and for Desktop Agents to provide custom implementations of them, or defer to reference implementations provided by the FDC3 Standard. This is achieved via to following messages:
```

Comment:
"via to following" should be "via the following"

Reply:
fixed in 976af80d5ba8d831e202aabcd37d1a7886e4c930